Sources/PayPalNativeCheckout/PayPalClient.swift

```diff
@@ -13,60 +9,67 @@ public class PayPalClient {
+        let nxoConfig = CheckoutConfig(
+            clientID: config.clientID, createOrder: nil, onApprove: nil,
+                    onShippingChange: nil, onCancel: nil, onError: nil, environment: config.environment.toNativeCheckoutSDKEnvironment())
+        self.nativeCheckout = NativeCheckout(nxoConfig: nxoConfig)
     }
 
-    init(config: CoreConfig, checkoutFlow: CheckoutProtocol.Type) {
+    init(config: CoreConfig, checkoutFlow: CheckoutProtocol) {
         self.config = config
-        self.CheckoutFlow = checkoutFlow
+        self.nativeCheckout = checkoutFlow
     }
 
     /// Present PayPal Paysheet and start a PayPal transaction
     /// - Parameters:
     ///   - request: the PayPalRequest for the transaction
     ///   - presentingViewController: the ViewController to present PayPalPaysheet on, if not provided, the Paysheet will be presented on your top-most ViewController
     ///   - completion: Completion block to handle buyer's approval, cancellation, and error.
-    public func start(request: PayPalRequest, presentingViewController: UIViewController? = nil) {
-        CheckoutFlow.set(config: config)
-
-        CheckoutFlow.start(
-            presentingViewController: presentingViewController,
-            createOrder: { order in
-                order.set(orderId: request.orderID)
-            },
-            onApprove: { approval in
-                self.notifySuccess(for: approval)
-            },
-            onCancel: {
-                self.notifyCancellation()
-            },
-            onError: { errorInfo in
-                self.notifyFailure(with: errorInfo)
-            }
-        )
+    public func start(presentingViewController: UIViewController?, orderID: String, deleagate: PayPalDelegate?){
+        self.delegate = deleagate
+        DispatchQueue.main.async {
+            self.nativeCheckout.start(
+                presentingViewController: presentingViewController,
+                createOrder: { order in
+                    order.set(orderId: orderID)
+                },
+                onApprove: { approval in
+                    self.notifySuccess(for: approval)
+                },
+                onShippingChange: { shippingChange, shippingChangeAction in
+                    self.notifyShippingChange(shippingChange: shippingChange, shippingChangeAction: shippingChangeAction)
+                },
+                onCancel: {
+                    self.notifyCancellation()
+                },
+                onError: {error in
+                    self.notifyFailure(with: error)
+                }
+            )
+        }
     }
 
-    private func notifySuccess(for approval: PayPalCheckoutApprovalData) {
-        let payPalResult = PayPalResult(orderID: approval.ecToken, payerID: approval.payerID)
-        delegate?.paypal(self, didFinishWithResult: payPalResult)
+    private func notifySuccess(for approval: PayPalCheckout.Approval) {
+        delegate?.paypal(didFinishWithResult: approval)
```

Comment:
Do we need these helper functions for one single line of code? Might be easier to just call `delegate?.paypal(didFinishWithResult: approval)` directly instead of using this helper function? Unless we think we will be utilizing these a lot.

Reply:
yeah just in case we move to different types helps to organize

Reply:
This is a good call out in general. I introduced these helper methods at some point. I was anxious about the strong implicit `self` retains in general. Like if NXO executes a callback but the View Controller is no longer on the screen? I was moving in the direction of `[weak self] in ... self?.notifySuccess()`.